[rayshawntan] iP

rayshawntan opened the pull request:
# Desolute Chatbot

> The only chatbot you need!

This chatbot helps you to remember the tasks you need to do, the deadlines you have and the events you need to attend! It's 

- Simple to use 
- Friendly to talk if you need a companion!
- Most importantly, it's **FREE**!!!!! 😀

All you need to do is,

1. Download from [here](https://github.com/rayshawntan/ip/releases/tag/Level-10)
2. Run it
3. Begin chatting!

Features:

- [x] Managing tasks
- [x] GUI for convenient use
- [ ] Styling of GUI

If you wanna learn to use it as a java programmer, this is the `main` method:
```
public static void main(String[] args) {
        new Main().run(args);
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

rayshawntan committed:
Change Duke class to chatBot class; Change output message to skeletal version of greetings and bot name

rayshawntan committed:
Modified code to echo commands, exits when user says bye and also included a try again message when none of the correct commands are given.

rayshawntan committed:
Modified chatBot to allow adding of Task and showing the list; added Task.java to store task info

rayshawntan committed:
Modified ChatBot.java to check for mark and unmark commands; Added instructions to Task.java to include mark and unmark

rayshawntan committed:
Rename chatBot.java to ChatBot.java

rayshawntan committed:
Rename unmarkDone function to unmarkedDone for both Chatbot.java and Task.java

rayshawntan committed:
Reorganised the mark and unmark into functions for easier referencing, Added missing line in list message in ChatBot.java

rayshawntan committed:
Added Deadline.java, Event.java, ToDo.java which are extensions of Task.java; updated ChatBot.java to include the 3 new types of Tasks; updated runtest.bat to take ChatBot class when running; updated input.txt and EXPECTED.TXT to check if test cases are correct.

rayshawntan committed:
Changed ChatBot.java to include DukeException.java; Updated input.txt and EXPECTED.TXT to include exception testing; Created DukeException to handle exceptions for ChatBot.java

rayshawntan committed:
Updated ChatBot.java to include delete function; Updated input.txt and EXPECTED.TXT for testing of delete function

rayshawntan committed:
Implement Level-7: Save

rayshawntan committed:
Implement Level-7: Save

rayshawntan committed:
Revert "Implement Level-7: Save"

This reverts commit 77cbec85eeeb692f1801fc0a94ce04977411d17e.

rayshawntan committed:
Merge branch 'branch-level-7'

rayshawntan committed:
Update ChatBot.java and list.txt

rayshawntan committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	data/list.txt
#	src/main/java/ChatBot.java

rayshawntan committed:
Rollback to Older Version
--------------
Errors in committing in master branch, hence rolling back to
attempt to roll to master branch

rayshawntan committed:
Merge remote-tracking branch 'origin/master'

rayshawntan committed:
Update ChatBot.java and Task.java

rayshawntan committed:
Empty java files

rayshawntan committed:
Update all java files

rayshawntan committed:
Update ChatBot.java and Task.java

rayshawntan committed:
Bug Fixes

rayshawntan committed:
Implement Level-8: Dates and Times

rayshawntan committed:
Merge branch 'branch-Level-8'

rayshawntan committed:
Implement A-More-OOP and A-Package

rayshawntan committed:
Bug Fix

rayshawntan committed:
Remove old ChatBot.java and Replace with ChatBot package

rayshawntan committed:
Merge branch 'branch-A-MoreOOP'

rayshawntan committed:
Merge branch 'master' into add-gradle-support

rayshawntan committed:
Implement A-Gradle

rayshawntan committed:
Merge branch 'add-gradle-support'

rayshawntan committed:
Implement A-JUnit

rayshawntan committed:
Merge branch 'branch-A-JUnit'

rayshawntan committed:
Implement A-JAR

rayshawntan committed:
Implement A-Javadocs

rayshawntan committed:
Merge branch 'branch-A-JavaDoc'

rayshawntan committed:
Implement A-CodingStandard

rayshawntan committed:
Merge branch 'branch-A-CodingStandard'

rayshawntan committed:
Implement Level-9

rayshawntan committed:
Merge branch 'branch-Level-9'

ChuanXinNg commented on src/main/java/rayshawn/chatbot/commands/FindCommand.java:
```diff
@@ -0,0 +1,49 @@
+        return new CommandResult(getTaskListCount(foundTasks), foundTasks);
+    }
+
+    private List<Task> getTasksWithNameContaining(String keyword) {
```
Perhaps you can name this method better such as getTasksByName.

ChuanXinNg commented on src/main/java/rayshawn/chatbot/messages/Messages.java:
```diff
@@ -0,0 +1,17 @@
+/**
+ * Container for user visible messages
+ */
+public class Messages {
```
Good practice on using constants for values that will be used repeatedly.

GohTengFong replied:
I agree! I did not even think of this!

GohTengFong replied:
I agree! I did not even think of this!

ChuanXinNg commented on src/main/java/rayshawn/chatbot/parser/Parser.java:
```diff
@@ -0,0 +1,197 @@
+import static rayshawn.chatbot.messages.Messages.INVALID_TASK_NUMBER_MESSAGE;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```
Good use of the java.util.regex package, it has helped to improve code readability and maintainability.

ChuanXinNg reviewed (approved):
I found your code easy to read and I discovered the use of packages that I never used before which are very useful for this project on managing code and improving code readability. You have done a great job, keep it up.

GohTengFong commented on src/main/java/rayshawn/chatbot/commands/Command.java:
```diff
@@ -0,0 +1,70 @@
+/**
+ * Represents an executable command.
+ */
+public class Command {
```
I love how you created many different classes for each Command. It really improves readability! However, would using enum suffice?

GohTengFong commented on src/main/java/rayshawn/chatbot/Main.java:
```diff
@@ -0,0 +1,88 @@
+     */
+    public void run(String[] args) {
+        start(args);
+        runCommandLoopUntilByeCommand();
```
Good job on naming this function! It was intuitive as a reader.

GohTengFong commented on src/main/java/rayshawn/chatbot/parser/Parser.java:
```diff
@@ -0,0 +1,197 @@
+
+        switch (commandWord) {
+
+        case ByeCommand.COMMAND_WORD:
```
Should there be a line break in-between each "case"?

GohTengFong reviewed (commented):
Overall, I think you did an amazing job with iP! I have actually learnt a lot by how you have extracted many different classes in order to achieve simplicity and readability.

rayshawntan committed:
Implement A-Checkstyle, Level-10 and A-Varargs

rayshawntan committed:
Merge branch 'branch-Level-10'

rayshawntan committed:
Apply Coding Standards

rayshawntan committed:
Implement A-Assertions

Adding assert statements to improve code reliability and catch
potential issues during debugging and testing. Assertions have
been added at critical points to check for specific conditions
and ensure the program behaves as expected.

rayshawntan committed:
Merge branch 'A-Assertions'

rayshawntan committed:
Implement A-CodeQuality

rayshawntan committed:
Merge branch 'branch-A-CodeQuality'

rayshawntan committed:
Implement BCD-Extensions, C-Update

Add the capability to update pre-existing tasks with new
information.

rayshawntan committed:
Merge pull request #2 from rayshawntan/BCD-Extensions

Implement BCD-Extensions, C-Update

rayshawntan committed:
Implement A-UserGuide

rayshawntan committed:
Merge branch 'branch-A-UserGuide'

rayshawntan committed:
Update README.md

rayshawntan committed:
Update Javadocs for UpdateCommand.java
